src/Nethermind/Nethermind.TxPool/BlobTxStorage.cs

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Db;
+using Nethermind.Int256;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.TxPool;
+
+public class BlobTxStorage : ITxStorage
+{
+    private readonly IDb _database;
+    private static readonly TxDecoder _txDecoder = new();
+
+    public BlobTxStorage()
+    {
+        _database = new MemDb();
+    }
+
+    public BlobTxStorage(IDb database)
+    {
+        _database = database ?? throw new ArgumentNullException(nameof(database));
+    }
+
+    public bool TryGet(ValueKeccak hash, [NotNullWhen(true)] out Transaction? transaction)
+        => TryDecode(_database.Get(hash.Bytes), out transaction);
+
+    private static bool TryDecode(byte[]? txBytes, out Transaction? transaction)
+    {
+        if (txBytes is not null)
+        {
+            RlpStream rlpStream = new(txBytes);
+            Address sender = rlpStream.DecodeAddress()!;
+            UInt256 timestamp = rlpStream.DecodeUInt256();
+            transaction = Rlp.Decode<Transaction>(rlpStream, RlpBehaviors.InMempoolForm);
+            transaction.SenderAddress = sender;
+            transaction.Timestamp = timestamp;
+            return true;
+        }
+
+        transaction = default;
+        return false;
+    }
+
+    public IEnumerable<Transaction> GetAll()
+    {
+        foreach (byte[] txBytes in _database.GetAllValues())
+        {
+            if (TryDecode(txBytes, out Transaction? transaction))
+            {
+                yield return transaction!;
+            }
```

Comment:
As the order doesn't matter, maybe the decoding of items could be parallelized?

Reply:
After refactoring, it is used only when recreating light blob collection on restart and whole recreating takes max 100-150ms. For now it is good enough, but parallelization is a good idea for further improvement in the future